python/cudf/tests/test_groupby.py

```diff
@@ -432,3 +430,26 @@ def test_list_of_series():
     pdg = pdf.groupby([pdf.x, pdf.y]).y.sum()
     gdg = gdf.groupby([gdf.x, gdf.y]).y.sum()
     assert_eq(pdg, gdg)
+
+
+def test_empty_groupby():
+    pdf = pd.DataFrame({'x': [], 'y': []})
+    gdf = cudf.from_pandas(pdf)
+    pdg = pdf.groupby(['x', 'y']).sum()
+    gdg = gdf.groupby(['x', 'y']).sum()
+    assert_eq(pdg, gdg)
+    pdg = pdf.groupby(['x', 'y']).agg('sum')
+    gdg = gdf.groupby(['x', 'y']).agg('sum')
+    assert_eq(pdg, gdg)
+    pdg = pdf.groupby(['y']).sum()
+    gdg = gdf.groupby(['y']).sum()
+    assert_eq(pdg, gdg)
+    pdg = pdf.groupby(['y']).agg('sum')
+    gdg = gdf.groupby(['y']).agg('sum')
+    assert_eq(pdg, gdg)
+    pdg = pdf.groupby(['x']).sum()
+    gdg = gdf.groupby(['x']).sum()
+    assert_eq(pdg, gdg)
+    pdg = pdf.groupby(['x']).agg('sum')
+    gdg = gdf.groupby(['x']).agg('sum')
+    assert_eq(pdg, gdg)
```

Comment:
For the future, I tend to parametrize these as follows:

```python
@pytest.mark.parametrize('func', [
    lambda df: df.groupby(['x', 'y']).sum(),
    ...
])
def test_empty_groupby(func):
    pdf = pd.DataFrame({'x': [], 'y': []})
    gdf = cudf.from_pandas(pdf)
    assert_eq(func(pdf), func(gdf))
```

Just a style thing though.  Nothing that should slow down this PR.